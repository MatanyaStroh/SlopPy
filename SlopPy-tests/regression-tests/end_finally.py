# handle finally opcode properly

# note that the 'finally' clause executes BEFORE the exception is caught
# at the top-level

def foo():
  try:
    raise Exception
  finally:
    print "FINALLY"

foo()
foo()
foo()

