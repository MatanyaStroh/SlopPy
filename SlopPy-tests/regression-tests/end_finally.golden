{NA <type 'exceptions.Exception'> Exception()
 Traceback (most recent call last):
  File "regression-tests/end_finally.py", line 8, in foo
    raise Exception

Locals: {}
}
{NA <type 'exceptions.Exception'> Exception()
 Traceback (most recent call last):
  File "regression-tests/end_finally.py", line 8, in foo
    raise Exception

Locals: {}
}
{NA <type 'exceptions.Exception'> Exception()
 Traceback (most recent call last):
  File "regression-tests/end_finally.py", line 8, in foo
    raise Exception

Locals: {}
}
